callback_plugins/default_plus_summary.py

```diff
@@ -0,0 +1,94 @@
+# vim: expandtab:tabstop=4:shiftwidth=4
+'''
+Ansible callback plugin.
+'''
+
+from ansible.plugins.callback.default import CallbackModule as CallbackModule_default
+from ansible import constants as C
+from ansible.utils.color import stringc
+
+
+class CallbackModule(CallbackModule_default):
+    '''
+    This is like the default callback plugin, but also stores results and
+    summarizes failures.
+    '''
+
+    CALLBACK_VERSION = 2.0
+    CALLBACK_TYPE = 'stdout'
```

Comment:
If you set CALLBACK_TYPE to 'aggregate', then you wouldn't need to replace the default stdout callback plugin.

Also, this and the module should live in a role.